pkg/cli/admin/nodeimage/basenodeimagecommand.go

```diff
@@ -87,7 +88,28 @@ func (c *BaseNodeImageCommand) getNodeJoinerPullSpec(ctx context.Context) error
 	}
 
 	// Extract the baremetal-installer image pullspec, since it
-	// provide the node-joiner tool.
+	// provides the node-joiner tool.
+
+	// First attempt to get the installer image from the configMap created by installer. This will work in disconnected environments.
+	installerImagesConfigMap, err := c.Client.CoreV1().ConfigMaps("openshift-config").Get(ctx, "installer-images", metav1.GetOptions{})
+	if err == nil {
+		images := installerImagesConfigMap.Data["images.json"]
+		if len(images) > 0 {
+			imageMap := make(map[string]string)
+			err := yaml.Unmarshal([]byte(images), &imageMap)
+			if err == nil {
+				installer, exists := imageMap["installer"]
+				if exists {
+					// Found pullSpec from configMap
+					c.log("installer pullspec obtained from installer-images configMap %s", installer)
+					c.nodeJoinerImage = installer
+					return nil
+				}
+			}
+		}
+	}
```

Comment:
If there's an error other than NotFound, we should at least log it.
And if it _is_ NotFound, we should probably log that we're trying the registry.

Reply:
Added logging.